src/api/OpenWith.js

```diff
@@ -20,6 +20,101 @@ class OpenWith extends Base {
 
         return `${this.getBaseApiUrl()}/files/${id}/open_with_integrations`;
     }
+
+    /**
+     * Gets Open With integration data
+     *
+     * @param {string} fileId - Box file ID
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
+     */
+    getOpenWithIntegrations(fileId: string, successCallback: Function, errorCallback: Function) {
+        this.get({
+            id: fileId,
+            successCallback: (openWithIntegrations) => {
+                const formattedOpenWithIntegrations = this.formatOpenWithData(openWithIntegrations);
```

Comment:
IMO, success callbacks/error should be sepearated out (for testability)

Reply:
@pramodsum @DanDeMicco how do you recommend passing in the success and error callbacks in this case. Unfortunately we need more than just the data returned by that API call.